src/amber/controller/param_validator.cr

```diff
@@ -0,0 +1,43 @@
+require "../support/predicates"
+
+# We are patching the String class and Number struct to extend the predicates
+# available this will allow to add friendlier methods for validation cases.
+class String
+  include Amber::Support::Predicates::String
+end
+
+abstract struct Number
+  include Amber::Support::Predicates::Number
+end
+
+module Amber
+  module Controller
+    module ParamValidator
+      # This is an validation result that will hold the success and
+      # hash for a given validation execution.
+      struct ValidationResult
+        getter? valid = false
+        getter params = {} of String => String
+
+        def initialize(@valid, @params); end
+      end
+
+      macro included
+        property? valid = false
+        property validated_params = {} of String => String
+
+        def validate
```

Comment:
@elorest I thought of extending the class instead of monkey patching. Now, what would be the design implications for extending or monkey patching this way? A couple of things that comes to mind is `params.validate {}` I guess this is similar to rails `params.required` the difference being is that we will be able to do the validations something that rails lack, the only framework that's doing this nicely is hanami. 

My personal preference is extending the **HTTP:: Params** class. Any thoughts?